packages/form-state-manager/src/utils/manager-api.ts

```diff
@@ -425,19 +428,25 @@ const createManagerApi: CreateManagerApi = ({
 
     // TODO modify all affected field state variables
     batch(() => {
+      const isEqualFn = state.fieldListeners[name]?.isEqual || defaultIsEqual;
+      const pristine = isEqualFn(value, state.fieldListeners[name]?.state?.meta?.initial || get(state.initialValues, name));
+
       setFieldState(name, (prevState) => ({
         ...prevState,
         meta: {
           ...prevState.meta,
-          pristine: false,
-          dirty: true
+          pristine,
+          dirty: !pristine
         },
         value
       }));
       state.pristine = false;
       state.dirty = true;
       validateField(name, value);
 
+      const validateFields = state.fieldListeners[name]?.validateFields;
+      validateFields && revalidateFields(validateFields);
```

Comment:
The validation must be run if the `validateFields` variable is undefined. Otherwise, it will break cross-field validation since the `allValues` object was changed.

Reply:
I don't get what you mean.

It has to run when `validateFields` is defined because via this option users tell form to validate other fields.

`allValues` is referenced to `state.values` so it will be always updated.

Reply:
Yes but all other fields must be revalidated on any field change by default. I am not sure if we do that at the moment. You can prove me wrong. I would add a test new for cross-field validation if possible. And we can be sure.

Reply:
No, only the changed field is validated.

`validateFields` is used to trigger validation of different field.

i.e.

```jsx
... name: 'name', validateFields: ['password'] ...
```

Password cannot contain name... so when you change `name` value, it will re-validate the `password`.

In this tests there is a case that checks only the changed field and the field in validateFields is triggered.

Reply:
You are right it does not revalidate all fields. It must have changed at some point :thinking: All right then nvm this.